### PR TITLE
Unit tests

### PR DESCRIPTION
**Pull request by khrystyna-sb**
separate network from ListView Controller
add some unit tests

### CODE REVIEW

File: CountriesTests/CountriesTableViewTests.swift [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+import XCTest
+@testable import Countries
+
+class CountriesTableViewTests: XCTestCase {
+
+    override func setUpWithError() throws {
+        // Put setup code here. This method is called before the invocation of each test method in the class.
+    }
+
```
Comment (Globalika) [inline CountriesTests/CountriesTableViewTests.swift]: this file is empty. do you need it?

File: CountriesUITests/CountriesUITests.swift [hunk 2]
```diff
@@ -1,42 +1,42 @@
...
+//
+//    func testLaunchPerformance() throws {
+//        if #available(macOS 10.15, iOS 13.0, tvOS 13.0, watchOS 7.0, *) {
+//            // This measures how long it takes to launch your application.
+//            measure(metrics: [XCTApplicationLaunchMetric()]) {
+//                XCUIApplication().launch()
+//            }
+//        }
+//    }
+//}
```
Comment (Globalika) [inline CountriesUITests/CountriesUITests.swift:42]: do you need these comment lines?

File: CountriesUITests/CountriesUITestsLaunchTests.swift [hunk 3]
```diff
@@ -1,32 +1,32 @@
...
+//
+//        // Insert steps here to perform after app launch but before taking a screenshot,
+//        // such as logging into a test account or navigating somewhere in the app
+//
+//        let attachment = XCTAttachment(screenshot: app.screenshot())
+//        attachment.name = "Launch Screen"
+//        attachment.lifetime = .keepAlways
+//        add(attachment)
+//    }
+//}
```
Comment (Globalika) [inline CountriesUITests/CountriesUITestsLaunchTests.swift:32]: saa

**Review by Globalika — CHANGES_REQUESTED** [review]
i left you comments. please resolve them

File: CountriesUITests/CountriesUITests.swift (hunk 2 above)
Comment (khrystyna-sb) [reply CountriesUITests/CountriesUITests.swift:42]: I commented them temporarily so that they do not interfere, maybe this code will be needed later

**Review by Globalika — APPROVED** [review]

File: CountriesUITests/CountriesUITests.swift (hunk 2 above)
Comment (apolwork) [reply CountriesUITests/CountriesUITests.swift:42]: Just drop it if you don't use it.
If you plan to return to it, I would suggest to store it in dedicated branch

File: Countries/ListViewController.swift [hunk 4]
```diff
@@ -82,7 +92,7 @@ class ListViewController: UITableViewController, UISearchBarDelegate, UISearchRe
         searchController.searchBar.tintColor = .black
     }
 
-    private func registerTableView() {
+    func registerTableView() {
```
Comment (ambrusha) [inline Countries/ListViewController.swift]: why `private` was removed?

File: Countries/NetworkManager.swift [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+        let continent = CountriesApiQuery.Data.Country.Continent(name: "testcontinent")
+        let language = CountriesApiQuery.Data.Country.Language(name: "testlanguage")
+        return CountriesApiQuery.Data.Country(code: "UA", name: "testname", capital: "testcapital", currency: "testcurrency", phone: "testphone", continent: continent, languages: [language])
+    }
+    
+    func getCountries(completion: @escaping (Result<[CountriesApiQuery.Data.Country]?, Error>) -> Void) {
+        completion(Result.success([getCountry(), getCountry()]))
+    }
+}
+
```
Comment (ambrusha) [inline Countries/NetworkManager.swift]: this must be moved to a separate TestClass

File: CountriesTests/CountriesTests.swift [hunk 6]
```diff
@@ -9,25 +9,80 @@ import XCTest
...
+    func testTableViewHeaderNotNill() {
+        let header = listVC.tableView(listVC.tableView, viewForHeaderInSection: 0)
+        XCTAssertNotNil(header)
     }
 
+    func testCurrentTypeOfCell() {
+        let cell = listVC.tableView(listVC.tableView, cellForRowAt: indexPath)
+        let isCurrentType = cell is CountryTableViewCell
+        XCTAssertTrue(isCurrentType)
+    }
```
Comment (ambrusha) [inline CountriesTests/CountriesTests.swift]: can be removed

File: CountriesTests/CountriesTests.swift [hunk 7]
```diff
@@ -9,25 +9,80 @@ import XCTest
...
+    
+    func testCurrentTypeOfHeader() {
+        let header = listVC.tableView(listVC.tableView, viewForHeaderInSection: 0)
+        let isCurrentType = header is CountryTableViewHeader
+        XCTAssertTrue(isCurrentType)
+    }
```
Comment (ambrusha) [inline CountriesTests/CountriesTests.swift]: can be removed

File: CountriesTests/CountriesTests.swift [hunk 8]
```diff
@@ -9,25 +9,80 @@ import XCTest
...
+    func testTableViewNumberOfRows() {
+        let numberOfRowsResult = listVC.tableView.numberOfRows(inSection: 0)
+        let filteredCountriesCount = listVC.filteredCountries.count
+        XCTAssertEqual(numberOfRowsResult, filteredCountriesCount)
+    }
+    
+    func testTableViewCellNotNil() {
+        let cell = listVC.tableView(listVC.tableView, cellForRowAt: indexPath)
+        XCTAssertNotNil(cell)
+    }
```
Comment (ambrusha) [inline CountriesTests/CountriesTests.swift]:
```suggestion
    func testTableViewCellNotNil() throws {
        let cell = try XCTUnwrap(listVC.tableView(listVC.tableView, cellForRowAt: indexPath) as? CountryTableViewCell)
        XCTAssertNotNil(cell)
    }
```

File: CountriesTests/CountriesTests.swift [hunk 9]
```diff
@@ -9,25 +9,80 @@ import XCTest
...
+    
+    func testTableViewHeaderNotNill() {
+        let header = listVC.tableView(listVC.tableView, viewForHeaderInSection: 0)
+        XCTAssertNotNil(header)
     }
```
Comment (ambrusha) [inline CountriesTests/CountriesTests.swift]:
```suggestion
    func testTableViewHeaderNotNill() throws {
        let header = try XCTUnwrap(listVC.tableView(listVC.tableView, viewForHeaderInSection: 0) as? CountryTableViewHeader)
        XCTAssertNotNil(header)
    }
```

File: Countries/ListViewController.swift (hunk 4 above)
Comment (khrystyna-sb) [reply Countries/ListViewController.swift]: I can't remember why it was removed, so I inserted it again

File: CountriesTests/CountriesTests.swift [hunk 10]
```diff
@@ -9,25 +9,69 @@ import XCTest
...
+        XCTAssertNotNil(result)
+    }
+    
+    func testFilterByContinentUppercase() {
+        let country = MockNetworkManager().getCountry()
+        let result = [country].filterByContinent(searchText: "TEST")
+        XCTAssertNotNil(result)
+    }
+    
+    func testFilterCountiesUppercase() {
```
Comment (ambrusha) [inline CountriesTests/CountriesTests.swift:54]: Counties -> Countries and other same typos

**Review by ambrusha — APPROVED** [review]
You still ignored using of `typealias` for all long apollo types...
But its your choice))

**Pull request merged**